### PR TITLE
Blade Heretic/Side knowledges improvements/fixes.

### PR DESCRIPTION
**Pull request by EnterTheJake**
## About The Pull Request

Blade Heretic has received a few changes.

The cost of crafting a Dark blade has been reduced in exchange for a lower blade capacity, The Dark blade itself has received a new sprite.

Updated Dark Blade

Realignment pulls you out stuns a bit faster and grants baton resistance while active.

 You may now infuse your blades with a (weaker) mansus grasp upon unlocking the ability to dual wield, they also gain increased demolition modifier.

Mawed Crucible now slowly refills and requires fewer organs to brew a potion; you may now use a charge to refill your eldritch flask.

The potion themselves have also received changes more on that below.

The cooldown on the cursed curio shield has been reduced.

Lionhunter's rifle no longer does increased damage on scoped targets, instead it marks them with Mansus grasp and teleports the heretic to them.

Lastly Blade ascension has been fixed, you once again get the Ring of Blades.

## Why It's Good For The Game

Oh boy, here we go.

# **Blade Heretic changes**

Blade Heretic sits in a pretty decent spot, I wouldn't call the path weak by any stretch of imagination, but there are few aspects that could be reasonably improved without changing the overall strength of the path significantly.

**Sundered Blade**
I  think these are too expensive to craft, especially compared to the other blades which require very basic materials.

 It's not uncommon to run into situations where you just cannot afford to make more than a set of blades, and i'd argue it's not fun for the crew to have their titanium or silver deposit drained every time a blade heretic rolls around.

As a solution, i'm halving the cost in exchange of lowering the cap from 5 to 4 blades.

**Realingment**

This spell is lowkey awful; 25 stamina regen per second really doesn't make much of a difference when you are getting chain batoned, I have footage of blade heretics dying to a single shove stun while this abilty  was active.

The stamina regen and reduce immobility timer has been buffed on top of granting baton resist so long as it stays active, so you can properly get in fighting position without getting constantly knocked down.

Mind you, It's still no CNS rebooter, so stuns will still yield a few seconds of vulnerability.

**Swift Blades reworked into Empowered Blades**

You may now use your Mansus grasp to infuse your Dark blades.

It comes with the tradeoff of losing the knockdown and the stamina damage, you still retain the backstab.

Video Demonstration: https://www.youtube.com/watch?v=9cO9BOD8Zz4

Dark Blades also gain increased demolition modifier.

Dual wielding puts the heretic in the annoying position of having to switch between the second Blade and an empty hand to use Mansus grasp.

Blade is supposed to be a master of melee combat, but they are still a dark mage, so why shouldn't they be able to infuse their blades?

It still comes with a tradeoff, I'd reckon super sweaty players will still want to hotswap, but hey, the option is there.

 The added demolition modifier is to provide Blade with some way of breaking in and out of places, given the path has no jaunts or utility whatsoever, this seems reasonable to me.

Lastly Malestrom of Silver finally works now; you once again get the blade aura upon ascending.

# **Side Knowledge changes**

**Mawed Crucible**

The crucible now passively refills, and has a special interaction to refill the Eldritch Flask, the potion themselves have received changes.

- Xray Potion: duration bumped from 60 to 90 seconds.

- Wall phasing potion: Duration bumped from 15 to 40 seconds, you may now recall to your original location at will.

- Potion of the Wounded soldier: Upon expiring, it heals your wounds and regrows missing limbs.

**Reasoning**: Let's be honest here, noone ever makes this thing, the cost of making 1 potion is exorbitant and the potion themselves are not even that good to begin with.

I'm not gonna explain every change in detail, but considering the crucible is one of the OG side knoweldges and you hardly hear anyone talk about it, we can safely give it a few buffs.

**Unfathomable Curio**

Cooldown on the shield has been halved.

**Reasoning**: discussed it with Rex (the guy who created it), 60 seconds for 1 block is a bit excessive , 30 seconds seems reasonable enough.

**Lionhunter's Rifle**

Made a bit easier to craft and maintain, it can now be stored in the vest slot of the Eldritch Robes.

The homing projectile now fully penetrates armor instead of having bonus damage;  it also marks the victim with Mansus grasp and teleports the Heretic directly to them, the homing on the projectile itself has been improved.

**Reasoning**: another side knowledge that sadly barely sees any play.

Frankly this gun just doesn't have a purpose to exist, long range weaponry don't really mix with Heretic toolkit all that well, as you want to get close to your target to drag em to the spook dimension, not snipe 'em from a distance

Lionhunter now works as an initiation tool, upon marking the target, the Heretic transforms into the fired bullet until it connects, applying mansus grasp on the victim.

Keep in mind you still need xray or thermals to use the rifle to its full potential, either from the Crucible or the ashen medallion.

Video Demonstration: https://www.youtube.com/watch?v=AXmidKrx-Fg

As a trade off, the damage has been halved from 60 to 30.

## Changelog

:cl:
balance: Sundered Blades now require 1 Titanium or Silver bar to craft and their capacity has been reduced to 4.
balance: Realignment pulls you out of stuns a bit faster and grants baton resist while active.
balance: Blade Heretic dual wielding now let's you infuse Your Dark Blades with a weaker mansus grasp and grants an increase in demolition modifier.
fix: Malestrom of Silver grants the ring of protective blades once again.
balance: Mawed Crucible requires 3 organs to brew one potion, passively refills overtime and can be used to refill the Eldritch Flask
balance: Brew of Crucible soul effect bumped to 40 seconds and can be ended early.
balance: Brew Of Dusk and Dawn effect bumped to 3 minutes.
balance: Brew of the  wounded soldier now offers a very minor passive heal and fully heals your wounds and limbs upon expiring.
balance: Cursed Curio shield now recharges faster.
balance: Lionhunter's rifle has been reworked, it now fits on the eldritch robes vest slots, it's cheaper to craft it and its ammunition and works as an initiation tool.
/:cl:

### CODE REVIEW

**Comment by EnterTheJake** [comment]
Drafting, as there are few issues I'd like to fix with the Lionhunter.

File: code/modules/antagonists/heretic/knowledge/side_blade_rust.dm [hunk 1]
```diff
@@ -73,12 +72,11 @@
 /datum/heretic_knowledge/rifle_ammo
 	name = "Lionhunter Rifle Ammunition"
 	desc = "Allows you to transmute 3 ballistic ammo casings (used or unused) of any caliber, \
-		including shotgun shot, with any animal hide to create an extra clip of ammunition for the Lionhunter Rifle."
+		including shotgun shot to create an extra clip of ammunition for the Lionhunter Rifle."
```
Comment (necromanceranne) [inline code/modules/antagonists/heretic/knowledge/side_blade_rust.dm:75]: It says shotgun 'shot' but the recipe calls for literally any ammo casing. This means anything from a 9mm to a foam dart. Maybe a 'bullet casing' is a better descriptor?

File: code/modules/antagonists/heretic/items/heretic_blades.dm [hunk 2]
```diff
@@ -68,11 +68,16 @@
 /obj/item/melee/sickly_blade/afterattack(atom/target, mob/user, click_parameters)
 	if(isliving(target))
 		SEND_SIGNAL(user, COMSIG_HERETIC_BLADE_ATTACK, target, src)
+	else
+		SEND_SIGNAL(user, COMSIG_HERETIC_BLADE_ATTACK_NON_LIVING, target, src)
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/items/heretic_blades.dm:72]: Don't add a second signal:
- Remove the `isliving` check, so the one signal is always sent
- Add `isliving` checks to existing implementations of the signal to keep parity

File: code/modules/antagonists/heretic/items/heretic_blades.dm [hunk 3]
```diff
@@ -137,6 +142,53 @@
...
+
+	if(!check_behind(user, target) || !isliving(target))
+		return
+
+	// We're officially behind them, apply effects
+	var/mob/living/living_target = target
+	living_target.AdjustParalyzed(1.5 SECONDS)
+	living_target.apply_damage(10, BRUTE, wound_bonus = CANT_WOUND)
+	living_target.balloon_alert(user, "backstab!")
+	playsound(get_turf(living_target), 'sound/items/weapons/guillotine.ogg', 100, TRUE)
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/items/heretic_blades.dm:176]:
```suggestion
	playsound(living_target, 'sound/items/weapons/guillotine.ogg', 100, TRUE)
```

File: code/modules/antagonists/heretic/items/heretic_blades.dm [hunk 4]
```diff
@@ -137,6 +142,53 @@
...
+		return
+	mark_to_apply.create_mark(user, target)
+
+	//Remove the infusion from any blades we own (and update their sprite)
+	for(var/obj/item/melee/sickly_blade/dark/to_infuse in user.get_all_contents_type(/obj/item/melee/sickly_blade/dark))
+		to_infuse.infused = FALSE
+		to_infuse.update_appearance(UPDATE_ICON)
+	user.update_held_items()
+
+	if(!check_behind(user, target) || !isliving(target))
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/items/heretic_blades.dm:168]: Shouldn't this isliving be wayyy earlier, like the second line of this proc? 

I don't think you can apply a mansus grasp mark to a non-living mob

File: code/modules/antagonists/heretic/items/heretic_blades.dm [hunk 5]
```diff
@@ -137,6 +142,53 @@
...
+	. = ..()
+	if(infused)
+		infused = FALSE
+		update_appearance(UPDATE_ICON)
+
+/obj/item/melee/sickly_blade/dark/update_icon_state()
+	. = ..()
+	if(infused)
+		icon_state = initial(icon_state) + "_infused"
+		inhand_icon_state = initial(icon_state) + "_infused"
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/items/heretic_blades.dm:188]: Prefer using / setting `base_icon_state` instead of using `initial`

File: code/modules/antagonists/heretic/items/hunter_rifle.dm [hunk 6]
```diff
@@ -134,6 +133,35 @@
...
+	if(isliving(firer))
+		var/mob/living/living_firer = firer
+		if(IS_HERETIC(living_firer))
+			living_firer.forceMove(src)
+			stored_mob = living_firer
+
+/obj/projectile/bullet/strilka310/lionhunter/Destroy()
+	if(stored_mob)
+		stored_mob.forceMove(loc)
+	return ..()
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/items/hunter_rifle.dm:150]: No side effects in destroy: This should be in on hit, impact, `on_range`.

Though, I am certain projectiles do not handle this cleanly at all, so we should probably keep this here, with a stack trace:
```suggestion
/obj/projectile/bullet/strilka310/lionhunter/Destroy()
	if(stored_mob)
		stack_trace("Lionhunter bullet qdel'd with its firer still inside!")
		stored_mob.forceMove(loc)
	return ..()
```

Comment (MrMelbert) [reply code/modules/antagonists/heretic/items/hunter_rifle.dm:150]: Also, the reference is unhandled. You will want to `null` the `stored_mob` var. 
You can do it easily with `Exited`
```dm
/obj/projectile/bullet/strilka310/lionhunter/Exited(atom/movable/gone, direction)
	. = ..()
	if(gone == stored_mob)
		stored_mob = null
```

File: code/modules/antagonists/heretic/items/hunter_rifle.dm [hunk 7]
```diff
@@ -134,6 +133,35 @@
...
+	. = ..()
+	if(!isliving(target))
+		return BULLET_ACT_HIT
+	var/mob/living/victim = target
+	var/mob/firing_mob = firer
+	if(IS_HERETIC_OR_MONSTER(victim) || !IS_HERETIC(firing_mob))
+		return BULLET_ACT_HIT
+
+	if(!victim.has_status_effect(/datum/status_effect/eldritch))
+		SEND_SIGNAL(firer, COMSIG_LIONHUNTER_ON_HIT, victim)
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/items/hunter_rifle.dm:162]: This is a strange signal, ideally this should be firing regardless of if the victim has the status effect or not.

Then the signal proc can say "don't do x if y"

File: code/modules/antagonists/heretic/knowledge/blade_lore.dm [hunk 8]
```diff
@@ -67,6 +67,17 @@
...
 	SIGNAL_HANDLER
 
+	if(!check_behind(source, target))
+		return
+
+	// We're officially behind them, apply effects
+	target.AdjustParalyzed(1.5 SECONDS)
+	target.apply_damage(10, BRUTE, wound_bonus = CANT_WOUND)
+	target.balloon_alert(source, "backstab!")
+	playsound(get_turf(target), 'sound/items/weapons/guillotine.ogg', 100, TRUE)
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/knowledge/blade_lore.dm:77]:
```suggestion
	playsound(target, 'sound/items/weapons/guillotine.ogg', 100, TRUE)
```

File: code/modules/antagonists/heretic/knowledge/blade_lore.dm [hunk 9]
```diff
@@ -83,24 +94,21 @@
...
+		var/mob/living/living_target = target
+		if(living_target.body_position == LYING_DOWN)
+			are_we_behind = TRUE
 
 	// Exceptions aside, let's actually check if they're, yknow, behind
 	var/dir_target_to_source = get_dir(target, source)
 	if(target.dir & REVERSE_DIR(dir_target_to_source))
 		are_we_behind = TRUE
 
 	if(!are_we_behind)
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/knowledge/blade_lore.dm:107]: You don't need this `are_we_behind` var since you separated it out into its own proc, you should just replace it with `return TRUE` / `return FALSE`

File: code/modules/antagonists/heretic/knowledge/blade_lore.dm [hunk 10]
```diff
@@ -330,7 +339,39 @@
...
+
+///Tries to infuse our held blade with our mansus grasp
+/datum/heretic_knowledge/blade_upgrade/blade/proc/on_grasp_cast(mob/living/carbon/cast_on)
+	SIGNAL_HANDLER
+
+	var/held_item = cast_on.get_active_held_item()
+	if(!istype(held_item, /obj/item/melee/sickly_blade/dark))
+		return NONE
+	var/obj/item/melee/sickly_blade/dark/held_blade = held_item
+	if(held_blade.infused == TRUE)
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/knowledge/blade_lore.dm:360]:
```suggestion
	if(held_blade.infused)
```

File: code/modules/antagonists/heretic/knowledge/side_blade_rust.dm (hunk 1 above)
Comment (MrMelbert) [reply code/modules/antagonists/heretic/knowledge/side_blade_rust.dm:75]: 1. The description is 
`Allows you to transmute 3 ballistic ammo casings (used or unused) of any caliber, including shotgun shot`
So you're right, 'bullet casing' is a better descriptor... it already uses it 😛 

2. Apparently .310 rounds are caseless. So it can't go infinite

File: code/modules/antagonists/heretic/magic/realignment.dm [hunk 11]
```diff
@@ -54,25 +54,28 @@
...
+	owner.add_traits(realignment_traits, id)
 	owner.add_filter(id, 2, list("type" = "outline", "color" = "#d6e3e7", "size" = 2))
 	var/filter = owner.get_filter(id)
 	animate(filter, alpha = 127, time = 1 SECONDS, loop = -1)
 	animate(alpha = 63, time = 2 SECONDS)
 	return TRUE
 
 /datum/status_effect/realignment/on_remove()
+	owner.remove_traits(realignment_traits, id)
 	REMOVE_TRAIT(owner, TRAIT_PACIFISM, id)
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/magic/realignment.dm:73]:
```suggestion
```

File: code/modules/antagonists/heretic/status_effects/buffs.dm [hunk 12]
```diff
@@ -4,16 +4,21 @@
...
+	///Action button to cancel the effect early
+	var/datum/action/cancel_crucible_soul/cancel_button
 
 /datum/status_effect/crucible_soul/on_apply()
 	to_chat(owner,span_notice("You phase through reality, nothing is out of bounds!"))
 	owner.alpha = 180
 	owner.pass_flags |= PASSCLOSEDTURF | PASSGLASS | PASSGRILLE | PASSMACHINE | PASSSTRUCTURE | PASSTABLE | PASSMOB | PASSDOORS | PASSVEHICLE
 	location = get_turf(owner)
+	cancel_button = new()
+	cancel_button.Grant(owner)
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/status_effects/buffs.dm:21]: You don't have to bother holding a ref, actions can do that for you

```suggestion

/datum/status_effect/crucible_soul/on_apply()
	to_chat(owner,span_notice("You phase through reality, nothing is out of bounds!"))
	owner.alpha = 180
	owner.pass_flags |= PASSCLOSEDTURF | PASSGLASS | PASSGRILLE | PASSMACHINE | PASSSTRUCTURE | PASSTABLE | PASSMOB | PASSDOORS | PASSVEHICLE
	location = get_turf(owner)
	var/datum/action/cancel_crucible_soul/cancel_button/cancel = new(src)
	cancel.Grant(owner)
```

File: code/modules/antagonists/heretic/status_effects/buffs.dm [hunk 13]
```diff
@@ -22,15 +27,29 @@
 	owner.pass_flags &= ~(PASSCLOSEDTURF | PASSGLASS | PASSGRILLE | PASSMACHINE | PASSSTRUCTURE | PASSTABLE | PASSMOB | PASSDOORS | PASSVEHICLE)
 	owner.forceMove(location)
 	location = null
+	cancel_button.Remove(owner)
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/status_effects/buffs.dm:30]:
```suggestion
```

File: code/modules/antagonists/heretic/status_effects/buffs.dm [hunk 14]
```diff
@@ -22,15 +27,29 @@
...
+	desc = "Use to end the blessing early"
+	button_icon = 'icons/obj/antags/eldritch.dmi'
+	button_icon_state = "crucible_soul"
+
+/datum/action/cancel_crucible_soul/Trigger(trigger_flags)
+	. = ..()
+	if(!.)
+		return
+	var/datum/status_effect/active_effect = owner.has_status_effect(/datum/status_effect/crucible_soul)
+	active_effect.Destroy()
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/status_effects/buffs.dm:46]: 1. No manual calls to `Destroy`!!
2. If you do above, `target` = `the effect`, so you can just do this
```suggestion
	qdel(target) // also qdels us
```

File: code/modules/antagonists/heretic/status_effects/buffs.dm [hunk 15]
```diff
@@ -58,12 +77,23 @@
 
 /datum/status_effect/marshal/on_remove()
 	owner.remove_movespeed_mod_immunities(id, /datum/movespeed_modifier/damage_slowdown)
+	if(!iscarbon(owner))
+		return
+	var/mob/living/carbon/drinker = owner
+	for(var/obj/item/bodypart/potentially_wounded in drinker.bodyparts)
+		for(var/datum/wound/found_wound in potentially_wounded.wounds)
+			found_wound.remove_wound()
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/status_effects/buffs.dm:85]:
```suggestion
	for(var/datum/wound/found_wound as anything in drinker.all_wounds)
		found_wound.remove_wound()
```

File: code/modules/antagonists/heretic/status_effects/buffs.dm [hunk 16]
```diff
@@ -161,7 +191,8 @@
 	if(QDELETED(src) || QDELETED(owner))
 		return
 
-	var/obj/effect/floating_blade/blade = new blade_type(get_turf(owner))
+	var/obj/effect/floating_blade/blade
+	blade = new blade_type(get_turf(owner))
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/status_effects/buffs.dm:195]: Why this change?

File: code/modules/antagonists/heretic/status_effects/buffs.dm [hunk 17]
```diff
@@ -22,15 +27,29 @@
...
+		return
+	var/datum/status_effect/active_effect = owner.has_status_effect(/datum/status_effect/crucible_soul)
+	active_effect.Destroy()
+
 // DUSK AND DAWN
 /datum/status_effect/duskndawn
 	id = "Blessing of Dusk and Dawn"
 	status_type = STATUS_EFFECT_REFRESH
-	duration = 60 SECONDS
+	duration = 3 MINUTES
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/status_effects/buffs.dm:52]: Also thinking about it now, this change + cheaper potions make me a little worry it's too easy to get 100% uptime on this.

File: code/modules/antagonists/heretic/structures/mawed_crucible.dm [hunk 18]
```diff
@@ -99,6 +115,16 @@
 		set_anchored(!anchored)
 		balloon_alert(user, "[anchored ? "":"un"]anchored")
 		return ITEM_INTERACT_SUCCESS
+	if(istype(tool, /obj/item/reagent_containers/cup/beaker/eldritch))
+		var/obj/item/reagent_containers/cup/beaker/eldritch/to_fill = tool
+		if(to_fill.reagents.total_volume >= to_fill.reagents.maximum_volume)
+			balloon_alert(user, "flask too full")
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/structures/mawed_crucible.dm:121]:
```suggestion
			balloon_alert(user, "flask is full!")
```

File: code/modules/antagonists/heretic/structures/mawed_crucible.dm [hunk 19]
```diff
@@ -99,6 +115,16 @@
...
+			return ITEM_INTERACT_SUCCESS
+		to_fill.reagents.add_reagent(/datum/reagent/eldritch, 50)
+		do_item_attack_animation(src, used_item = tool)
+		current_mass--
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/structures/mawed_crucible.dm:125]: Feel like this should use all the mass, since it's equivalent to "making a new potion"

File: code/modules/antagonists/heretic/structures/mawed_crucible.dm [hunk 20]
```diff
@@ -11,15 +11,28 @@
...
 
 /obj/structure/destructible/eldritch_crucible/Initialize(mapload)
 	. = ..()
 	break_message = span_warning("[src] falls apart with a thud!")
+	START_PROCESSING(SSobj, src)
+
+/obj/structure/destructible/eldritch_crucible/process(seconds_per_tick)
+	if(world.time < last_refill + refill_cooldown || current_mass >= max_mass)
+		return
+	last_refill = world.time
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/structures/mawed_crucible.dm:32]: Use cooldown defines `COOLDOWN_DECLARE(refill_cd)` `COOLDOWN_START(src, refill_cd, 30 SECONDS)`

File: code/modules/antagonists/heretic/knowledge/blade_lore.dm [hunk 21]
```diff
@@ -372,13 +413,20 @@
...
 	// Perform the offhand attack
 	blade.melee_attack_chain(source, target)
 	// Restore the force.
 	blade.force = last_weapon_force
 
+///Modifies our blade demolition modifier so we can take down doors with it
+/datum/heretic_knowledge/blade_upgrade/blade/proc/on_blade_equipped(mob/user, obj/item/equipped, slot)
+	SIGNAL_HANDLER
+	if(istype(equipped, /obj/item/melee/sickly_blade/dark))
+		equipped.demolition_mod = 1.5
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/knowledge/blade_lore.dm:428]: You don't have to change it in this pr but I would love for us to have a framework for "knowledge that changes your blades in some way", this is kind of jank (but acceptable)

File: code/modules/antagonists/heretic/status_effects/buffs.dm (hunk 15 above)
Comment (MrMelbert) [reply code/modules/antagonists/heretic/status_effects/buffs.dm:85]: I also think this could use a visual or audio effect, kind of like how lings go "crunch" when they regrow stuff. 

Just, instantly healing all wounds and regaining all limbs with no pizzazz feels off.

File: code/modules/antagonists/heretic/status_effects/buffs.dm (hunk 16 above)
Comment (Xander3359) [reply code/modules/antagonists/heretic/status_effects/buffs.dm:195]: Fixes a runtime causing the ascension to not spawn the ring of knives.

File: code/modules/antagonists/heretic/knowledge/side_blade_rust.dm (hunk 1 above)
Comment (necromanceranne) [reply code/modules/antagonists/heretic/knowledge/side_blade_rust.dm:75]: 1. Ah, yeah I realized it was already mentioning that. Then in that case I think it's probably just unnecessary to mention shotgun shells, I think people would probably already interpret it as valid. Wasn't this at one point 'improvised shot' or something?

2 & 3. I don't believe it has to be empty casings, does it? Empty casings are not a separate type of item. I see the blacklist now but it doesn't have a check for if the bullet is loaded or not as far as I can tell. (it probably shouldn't as that would be arbitrarily more difficult to reload the gun if you needed a separate gun to unload all your bullets with)

**Comment by necromanceranne** [comment]
I already knew most of this was in the works, but I'm very personally happy that blade heretics can channel their grasp into their blade, that was something I was personally thinking could be an interesting addition since they're never usually going to be without a blade in their offhand pretty much ever once they have Swift Blades. It was an extremely awkward inventory fumble they constantly had to do every time they wanted to maximize their combat potential.

Given that it is a much weaker version as well, this feels like a completely reasonable way of handling it as well.

**Comment by Xander3359** [comment]
> I already knew most of this was in the works, but I'm very personally happy that blade heretics can channel their grasp into their blade, that was something I was personally thinking could be an interesting addition since they're never usually going to be without a blade in their offhand pretty much ever once they have Swift Blades. It was an extremely awkward inventory fumble they constantly had to do every time they wanted to maximize their combat potential.
> 
> Given that it is a much weaker version as well, this feels like a completely reasonable way of handling it as well.

Yeah it doesnt have the stun from funny handing someone but you do get the mark which gives you a protective blade with the 2nd hit activating the mark.

**Comment by carlarctg** [comment]
while we're talking about the grasp i always disliked that it had a cool  'lock them in your room' effect that goes away the second you start fighting them. that always felt to me like a letdown and a waste. maybe turn it into a targeted spell instead?

**Comment by EnterTheJake** [comment]
> while we're talking about the grasp i always disliked that it had a cool 'lock them in your room' effect that goes away the second you start fighting them. that always felt to me like a letdown and a waste. maybe turn it into a targeted spell instead?

Isn't that what lock already gets?

**Comment by carlarctg** [comment]
sure. it still fits with the theme of blade though

File: code/modules/antagonists/heretic/status_effects/buffs.dm (hunk 17 above)
Comment (EnterTheJake) [reply code/modules/antagonists/heretic/status_effects/buffs.dm]: Made it 90 seconds, shouldn't guarantee 100% uptime but grant a little extra leeway.

**Comment by EnterTheJake** [comment]
Ready for review.

Lionhunter homing projectile takes a bit less to home in, it now fully penetrates armor (30 brute with no armor penetration would pretty much tickle a sec off otherwise) and fits in the vest slot of the Eldritch robes.

File: code/modules/antagonists/heretic/items/hunter_rifle.dm [hunk 22]
```diff
@@ -134,6 +134,46 @@
...
+	var/mob/stored_mob
+
+/obj/projectile/bullet/strilka310/lionhunter/fire(angle, atom/direct_target)
+	. = ..()
+	if(!isliving(firer) || !isliving(original))
+		return
+	var/mob/living/living_firer = firer
+	if(IS_HERETIC(living_firer))
+		living_firer.swap_hand() //Forcibly unscope
+		living_firer.swap_hand()
```
Comment (Ghommie) [inline code/modules/antagonists/heretic/items/hunter_rifle.dm:147]: I feel like we as well make it so the scope component unscopes if the mob is moved on a location that isn't a turf at this point.

File: code/modules/antagonists/heretic/items/hunter_rifle.dm [hunk 23]
```diff
@@ -134,6 +134,46 @@
...
+/obj/projectile/bullet/strilka310/lionhunter/on_range()
+	if(stored_mob)
+		stored_mob.forceMove(loc)
+		stored_mob = null
+	return ..()
+
+/obj/projectile/bullet/strilka310/lionhunter/on_hit(atom/target, blocked, pierce_hit)
+	if(stored_mob) //Pretty important to get our mob out of the bullet
+		stored_mob.forceMove(loc)
+		stored_mob = null
```
Comment (Ghommie) [inline code/modules/antagonists/heretic/items/hunter_rifle.dm:160]:
```suggestion

/obj/projectile/bullet/Exited(atom/movable/gone)
	if(gone == stored_mob)
		stored_mob = null
	return ..()

/obj/projectile/bullet/strilka310/lionhunter/on_range()
	stored_mob?.forceMove(loc)
	return ..()

/obj/projectile/bullet/strilka310/lionhunter/on_hit(atom/target, blocked, pierce_hit)
	stored_mob?.forceMove(loc) //Pretty important to get our mob out of the bullet
```

File: code/modules/antagonists/heretic/knowledge/blade_lore.dm [hunk 24]
```diff
@@ -67,40 +67,45 @@
...
+		return
+
+	// We're officially behind them, apply effects
+	target.AdjustParalyzed(1.5 SECONDS)
+	target.apply_damage(10, BRUTE, wound_bonus = CANT_WOUND)
+	target.balloon_alert(source, "backstab!")
+	playsound(target, 'sound/items/weapons/guillotine.ogg', 100, TRUE)
+
+///Checks to see if `atom/source` is behind `atom/target`
+/proc/check_behind(atom/source, atom/target)
```
Comment (Ghommie) [inline code/modules/antagonists/heretic/knowledge/blade_lore.dm:80]: i think this should be moved to the helpers directory, perhaps in the game.dm file.

File: code/modules/antagonists/heretic/status_effects/buffs.dm [hunk 25]
```diff
@@ -58,12 +75,24 @@
 
 /datum/status_effect/marshal/on_remove()
 	owner.remove_movespeed_mod_immunities(id, /datum/movespeed_modifier/damage_slowdown)
+	if(!iscarbon(owner))
+		return
+	var/mob/living/carbon/drinker = owner
+	for(var/obj/item/bodypart/potentially_wounded in drinker.bodyparts)
```
Comment (Ghommie) [inline code/modules/antagonists/heretic/status_effects/buffs.dm:81]:
```suggestion
	for(var/obj/item/bodypart/potentially_wounded as anything in drinker.bodyparts)
```

File: code/modules/antagonists/heretic/status_effects/buffs.dm [hunk 26]
```diff
@@ -58,12 +75,24 @@
...
+		for(var/datum/wound/found_wound in potentially_wounded.wounds)
```
Comment (Ghommie) [inline code/modules/antagonists/heretic/status_effects/buffs.dm:82]:
```suggestion
		for(var/datum/wound/found_wound as anything in potentially_wounded.wounds)
```

File: code/modules/antagonists/heretic/status_effects/buffs.dm [hunk 27]
```diff
@@ -58,12 +75,24 @@
...
+			found_wound.remove_wound()
+	if(length(drinker.get_missing_limbs()) >= 1)
```
Comment (Ghommie) [inline code/modules/antagonists/heretic/status_effects/buffs.dm:84]:
```suggestion
	if(length(drinker.get_missing_limbs()))
```
The if statements won't pass if length() returns zero anyway, and it cannot return anything below that.

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/heretic/items/hunter_rifle.dm (hunk 22 above)
Comment (Xander3359) [reply code/modules/antagonists/heretic/items/hunter_rifle.dm:147]: That sounds reasonable

File: code/modules/antagonists/heretic/items/hunter_rifle.dm [hunk 28]
```diff
@@ -134,6 +134,48 @@
...
+	var/mob/stored_mob
+
+/obj/projectile/bullet/strilka310/lionhunter/fire(angle, atom/direct_target)
+	. = ..()
+	if(!isliving(firer) || !isliving(original))
+		return
+	var/mob/living/living_firer = firer
+	if(IS_HERETIC(living_firer))
+		//living_firer.swap_hand() //Forcibly unscope
+		//living_firer.swap_hand()
```
Comment (Ghommie) [inline code/modules/antagonists/heretic/items/hunter_rifle.dm:147]: Remove these two lines now.

Comment (EnterTheJake) [reply code/modules/antagonists/heretic/items/hunter_rifle.dm]: Done

**Review by Ghommie — APPROVED** [review]
